[APISMART-1386] add new mutation for update Fahrzeugkauf (#53)
europace/kex-vorgang-update-api: README.md

* The Datenkontext (TESTUMGEBUNG|ECHTGESCHAEFT) has to be allowed for the authenticated user.
* Values of fields, which do not have a default value and are not specified in the request, will be deleted.

### Update Fahrzeugkauf

#### Hints

> * The `finanzierungszweck` of the Vorgang has to be `FAHRZEUGKAUF` otherwise values of any fields in your request won't be saved. Additionally if the `finanzierungszweck` of the Vorgang isn't `FAHRZEUGKAUF` and you try to update the Vorgang, you will receive an error message.

Reviewer: Ich würde gern won't, isn't u.ä. vermeiden und das immer ausschreiben:

```suggestion
* The `finanzierungszweck` of the Vorgang has to be `FAHRZEUGKAUF` otherwise values of any fields in your request will not be saved. 
```

Oder alternativ:
```suggestion
* The `finanzierungszweck` of the Vorgang has to be `FAHRZEUGKAUF` otherwise the update will be ignored.
```

Es ist ja keine Fehlermeldung, sondern "nur" eine `message`. Daher würde ich den Rest einfach weglassen, wir haben auch an anderen Stellen nicht genauer erklärt, wann Messages zurückkommen.

> * Changing the `finanzierungszweck` to another than `FAHRZEUGKAUF` will delete all values of any previously stored fields.

Reviewer 2: ```suggestion
* Changing the `finanzierungszweck` to another value will delete all values of any previously stored fields.
```

Mit dem Satz oben drüber, weiß man ja bereits, dass der Finanzierungszweck `FAHRZEUGKAUF`  ist. Bei der (Fake) "Änderung" von `FAHRZEUGKAUF`  zu `FAHRZEUGKAUF`, würde ich kein besonderes Verhalten erwarten und es deshalb auch nicht explizit erwähnen.

Falls das `FAHRZEUGKAUF` doch explizit erwähnt werden soll, fehlt mir irgendwie ein Fülltwort zwischen "another" und "than"

Author: @XiangRongLin Ja stimmt werde dein Vorschlag gerne übernehmen

Reviewer 2: @Yildiz-Ercan Kannst du diese Änderung noch pushen, dann passt es für mich

> * Changing the `finanzierungszweck` to another than `FAHRZEUGKAUF` will delete all values of any previously stored fields.

Reviewer: ```suggestion
* Changing the `finanzierungszweck` to something other than `FAHRZEUGKAUF` will delete a previously saved `fahrzeugkauf`.
```

> * The fields `ps` and `kw` don't have default values and also these fields won't be converted.

Reviewer 2: Ist das nötig? Normalerweise gehe ich nicht von so einer Funktionalität aus und würde es deshalb auch nicht explizit erwähnen.

**updateFahrzeugkauf** ( vorgangsnummer: String!, fahrzeugkauf: [Fahrzeugkauf](#fahrzeugkauf)! ) -> [BasicResponse](#basicresponse)!

> This mutation is for updating the [Fahrzeugkauf](#fahrzeugkauf) of a Vorgang.

#### Example

##### POST Request

    POST https://kex-vorgaenge.ratenkredit.api.europace.de/vorgaenge
    Authorization: Bearer xxxx
    Content-Type: application/json

    {
      "query": "mutation fahrzeugkauf($vorgangsnummer: String!) {  
        updateFahrzeugkauf(vorgangsnummer: $vorgangsnummer, fahrzeugkauf: {

>         modell: "E230",

Reviewer 2: Sollten die "" nicht escaped werden? Ansonsten ist das doch kein valides JSON

Author: @XiangRongLin Ich habe auch nochmal nachgeschaut du hast recht das ist kein valides JSON, eher zur Veranschaulichung. Mit entfernen der Zeilenumbrüche (mit wird es ziemlich hässlich) würde das ungefähr so aussehen:

{
   "{\"query\": \"mutation fahrzeugkauf($vorgangsnummer:String!) { updateFahrzeugkauf(vorgangsnummer: $vorgangsnummer, fahrzeugkauf: { modell: \"E230\", marke: \"Mercedes-Benz\", kaufpreis: 59990.99, erstzulassungsdatum: \"2020-01-30\", laufleistung: 89001, kw: 210, beglicheneKosten: 25990.99, ps: 333, anbieter: HAENDLER}){ messages }}\",\"variables\": {\"vorgangsnummer\": \"ABC123\"}}":null
}

Würde ich trotzdem erstmal so beibehalten für den PR und deinen Einwand zur Diskussion stellen, weil das eine Entscheidung ist die wir gemeinsam treffen sollten und wenn wir das nur an der einen Stelle anpassen ist es auch nicht erledigt.

Reviewer: Das Problem haben wir auf jeden Fall auch an den anderen Stellen, da müssen wir wohl auch noch ran. Aber hier an dieser Stelle würde ich das Beispiel komplett rauslassen, wir wollten davon eigentlich weg kommen, weil es die Doku  unübersichtlicher und größer macht. Wenn man einige wenige Beispiele hat, kann man den Rest davon ableiten.
